File: domain-driven-design/node-web/src/infrastructure/repository/repository.js
// src/infrastructure/repository/repository.js
// 模拟通用接口，JS因为没有接口字段，故此文件无用
class Repository {
  // 保存实体
  async save(entity) {
    throw new Error('save 方法未实现');
  }

  // 根据ID查找实体
  async findByID(id) {
    throw new Error('findByID 方法未实现');
  }

  // 查找所有实体
  async findAll() {
    throw new Error('findAll 方法未实现');
  }

  // 删除实体
  async delete(id) {
    throw new Error('delete 方法未实现');
  }
}

module.exports = Repository;

File: domain-driven-design/go-web/internal/domain/repository/repository.go
package repository

// Repository 通用仓储接口，定义了常见的 CRUD 操作
type Repository[T any] interface {
  Save(entity *T) error         // 保存实体
  FindByID(id int64) (*T, error)  // 根据ID查找实体
  FindAll(userId int) ([]*T, error)       // 查找所有实体
  Delete(id int64) error          // 删除实体
}
